### PR TITLE
[5.4] add default param to collection's when() method

### PR DESCRIPTION
**Pull request by tomschlick**
This brings the changes made in #17917 up to date with my original changes to the eloquent query builder made in #15428

### CODE REVIEW

File: src/Illuminate/Support/Collection.php [hunk 1]
```diff
@@ -315,10 +315,12 @@ public function filter(callable $callback = null)
      * @param  callable  $callback
      * @return mixed
      */
-    public function when($value, callable $callback)
+    public function when($value, callable $callback, callable $default = null)
     {
         if ($value) {
             return $callback($this);
+        } elseif ($default) {
+            return $default($this);
```
Comment (ntzm) [inline src/Illuminate/Support/Collection.php]: this `elseif` can just be an `if`

File: src/Illuminate/Support/Collection.php [hunk 2]
```diff
@@ -315,10 +315,12 @@ public function filter(callable $callback = null)
      * @param  callable  $callback
```
Comment (ntzm) [inline src/Illuminate/Support/Collection.php]: docblock needs updating as well

File: src/Illuminate/Support/Collection.php (hunk 1 above)
Comment (tomschlick) [inline src/Illuminate/Support/Collection.php]: Gonna leave that as is. Thats what we currently use for the eloquent QB and it works just fine for now

File: src/Illuminate/Support/Collection.php (hunk 2 above)
Comment (tomschlick) [inline src/Illuminate/Support/Collection.php]: done

**Pull request merged**